ray-operator/test/e2e/rayjob_test.go

```diff
@@ -245,4 +246,51 @@ env_vars:
 		g.Expect(GetRayJob(test, rayJob.Namespace, rayJob.Name)).
 			To(WithTransform(RayJobReason, Equal(rayv1.DeadlineExceeded)))
 	})
+
+	test.T().Run("RayJob should be created, but not updated when managed externaly", func(_ *testing.T) {
+		// RayJob
+		rayJobAC := rayv1ac.RayJob("managed-externally", namespace.Name).
+			WithSpec(rayv1ac.RayJobSpec().
+				WithRayClusterSpec(newRayClusterSpec(mountConfigMap[rayv1ac.RayClusterSpecApplyConfiguration](jobs, "/home/ray/jobs"))).
+				WithEntrypoint("python /home/ray/jobs/counter.py").
+				WithRuntimeEnvYAML(`
+env_vars:
+  counter_name: test_counter
+`).
+				WithShutdownAfterJobFinishes(true).
+				WithSubmitterPodTemplate(jobSubmitterPodTemplateApplyConfiguration()).
+				WithManagedBy(rayv1.MultiKueueController))
+
+		rayJob, err := test.Client().Ray().RayV1().RayJobs(namespace.Name).Apply(test.Ctx(), rayJobAC, TestApplyOptions)
+		g.Expect(err).NotTo(HaveOccurred())
+		test.T().Logf("Created RayJob %s/%s successfully", rayJob.Namespace, rayJob.Name)
+
+		// Refresh the RayJob status and assert it has not been updated
+		g.Eventually(RayJob(test, rayJob.Namespace, rayJob.Name)).
+			Should(WithTransform(RayJobDeploymentStatus, Equal(rayv1.JobDeploymentStatusNew)))
+
+		// Assert the associated RayCluster has not beed created
+		rcList, err := test.Client().Ray().RayV1().RayClusters(rayJob.Namespace).List(test.Ctx(), metav1.ListOptions{})
+		g.Expect(err).NotTo(HaveOccurred())
+		for _, rc := range rcList.Items {
+			g.Expect(rc.Name).NotTo(HaveSuffix(*rayJobAC.Name))
+		}
+
+		// Assert the submitter Job has not been created
+		g.Eventually(Jobs(test, namespace.Name)).Should(BeEmpty())
+
+		// Should not to be able to change managedBy field as it's immutable
+		rayJobAC.Spec.WithManagedBy(rayv1.RayJobController)
+		_, err = test.Client().Ray().RayV1().RayJobs(namespace.Name).Apply(test.Ctx(), rayJobAC, TestApplyOptions)
+		g.Expect(err).To(HaveOccurred())
+		g.Eventually(RayJob(test, *rayJobAC.Namespace, *rayJobAC.Name)).
+			Should(WithTransform(RayJobManagedBy, Equal(ptr.To(rayv1.MultiKueueController))))
+		g.Eventually(RayJob(test, *rayJobAC.Namespace, *rayJobAC.Name)).
+			Should(WithTransform(RayJobDeploymentStatus, Equal(rayv1.JobDeploymentStatusNew)))
```

Comment:
I think we can drop this as this is checked already ones

Reply:
ok